daemon/daemon.go

```diff
@@ -485,6 +533,14 @@ func (d *Daemon) Start() {
 	// the loop runs in its own goroutine
 	d.overlord.Loop()
 
+	// see if we can go into socket activation mode
+	go func() {
+		time.Sleep(goSocketActivateWait)
```

Comment:
this looks ok, but wondering if we have the reverse problem this way, situations in which we would stay up but shouldn't because in the end no snap were installed.  I suppose the only kind of change like would be a configuration change for core?

Reply:
Yes, I think you are right, the code should not check this only once but a couple of times, I will look into this.

Reply:
As an experiment I did a version of this code that uses a manager to hook into the ensure loop in https://github.com/snapcore/snapd/pull/5757 - feedback welcome. Happy to close it if it feels too much out of place as a manager, I think I will still try to move the logic into its own package.